### PR TITLE
conf-snappy: add include dir on FreeBSD

### PR DESCRIPTION
**Pull request by jmid**
On FreeBSD this fails with https://freebsd.check.ci.dev/log/1713371636-f089d8564db935c703bea70a4b3d8335ffa1bef3/4.14.1/bad/conf-snappy.1
```
# test.cpp:1:10: fatal error: 'snappy.h' file not found
# #include <snappy.h>
#          ^~~~~~~~~~
# 1 error generated.
```

The PR therefore adds a (standard) FreeBSD include directory.

### CODE REVIEW

File: packages/conf-snappy/conf-snappy.1/opam [hunk 1]
```diff
@@ -3,7 +3,8 @@ maintainer: "gabriel.scherer@inria.fr"
 homepage: "https://google.github.io/snappy/"
 bug-reports: "https://github.com/ocaml/opam-repository/issues"
 build: [
-  ["c++" "test.cpp" "-lsnappy" "-std=c++11"]
+  ["c++" "test.cpp" "-lsnappy" "-std=c++11"] {os != "freebsd"}
+  ["c++" "test.cpp" "-I/usr/local/include" "-lsnappy" "-std=c++11"] {os = "freebsd"}
```
Comment (mseri) [inline packages/conf-snappy/conf-snappy.1/opam:7]: Does not seem to work. In this case, I think you could also do
```suggestion
  ["c++" "test.cpp"
   "-I/usr/local/include" {os = "freebsd"}
   "-lsnappy" "-std=c++11"] 
```

**Comment by jmid** [comment]
Progress - now the header is found on FreeBSD - but the library isn't:
```
### output ###
# ld: error: unable to find library -lsnappy
# c++: error: linker command failed with exit code 1 (use -v to see invocation)
```

- 6fd6708 sets a FreeBSD library path
- acc4d6b silences Oracle CI errors
- 092e762 adds an include and library path since the header wasn't being found on macOS ARM64 either:
```
### output ###
# test.cpp:1:10: fatal error: 'snappy.h' file not found
# #include <snappy.h>
#          ^~~~~~~~~~
# 1 error generated.
```

**Comment by mseri** [comment]
Now only macos fails. I am inclined to merge it as is, what do you think?

**Comment by jmid** [comment]
I would like to fix the remaining macOS ARM64 failure first. :pray:
I suspect this is just missing the `sh -c` invocation you've suggested in the other PRs? :thinking:

**Pull request merged**